src/postgres_mcp/dta/sql_driver.py

```diff
@@ -29,39 +32,73 @@ def __init__(self, conn: Any = None, engine_url: str | None = None):
         if conn:
             self.conn = conn
         elif engine_url:
-            self.conn = psycopg2.connect(engine_url)
+            # Don't connect here since we need async connection
+            self.engine_url = engine_url
+            self.conn = None
         else:
             raise ValueError("Either conn or engine_url must be provided")
 
-    def execute_query(
-        self, query: str, force_readonly: bool = True
+    async def connect(self):
+        if self.conn is not None:
+            return
+        if self.engine_url:
+            self.conn = await psycopg.AsyncConnection.connect(self.engine_url)
+        else:
+            raise ValueError(
+                "Connection not established. Either conn or engine_url must be provided"
+            )
+
+    async def execute_query(
+        self,
+        query: LiteralString,
+        params: list[Any] | None = None,
+        force_readonly: bool = True,
     ) -> Optional[List[RowResult]]:
         """
         Execute a query and return results.
 
         Args:
             query: SQL query to execute
+            params: Query parameters
+            force_readonly: Whether to enforce read-only mode
 
         Returns:
             List of RowResult objects or None on error
         """
         try:
-            with self.conn.cursor(cursor_factory=RealDictCursor) as cursor:
+            await self.connect()
+            if self.conn is None:
+                raise ValueError("Connection not established")
+            async with self.conn.cursor(row_factory=dict_row) as cursor:
                 # Start read-only transaction
                 if force_readonly:
-                    cursor.execute("BEGIN TRANSACTION READ ONLY")
+                    await cursor.execute("BEGIN TRANSACTION READ ONLY")
                 try:
-                    cursor.execute(query)
+                    if params:
+                        await cursor.execute(query, params)
+                    else:
+                        await cursor.execute(query)
+
+                    # For multiple statements, move to the last statement's results
+                    while cursor.nextset():
+                        pass
+
                     if cursor.description is None:  # No results (like DDL statements)
```

Comment:
At some point we may want to change the interface to return the results of all statements.